hcipy/coronagraphy/multi_scale.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+from scipy.signal import windows
+
+from ..optics import OpticalElement, Apodizer, Wavefront
+from ..propagation import FraunhoferPropagator
+from ..field import make_focal_grid
+from ..fourier import FastFourierTransform, MatrixFourierTransform, FourierFilter
+
+class MultiScaleCoronagraph(OpticalElement):
+    '''A phase mask coronagraph.
+
+    This :class:`OpticalElement` simulates the propagation of light through
+    a phase mask in the focal plane. To resolve the singularity of the
+    phase plate, a multi-scale approach is made. Discretisation errors made at
+    a certain level are corrected by the next level with finer sampling.
+
+    Parameters
+    ----------
+    input_grid : Grid
+        The grid on which the incoming wavefront is defined.
+    complex_mask : field_generator
+        The complex focal-plane phase mask.
+    lyot_stop : Field or OpticalElement
+        The Lyot stop for the coronagraph. If it's a Field, it is converted to an
+        OpticalElement for convenience. If this is None (default), then no Lyot stop is used.
+    q : scalar
+        The minimum number of pixels per lambda/D. The number of levels in the multi-scale
+        Fourier transforms will be chosen to reach at least this number of samples. The required
+        q for a high-accuracy vortex coronagraph for example depends on the charge of the vortex. For charge 2,
```

Comment:
@syhaffert This is one of the instances where the vortex is still explicitly stated in the docstring of the now generalized multi-scale coronagraph. I see this as an example to illustrate the parameter `q` though, so I purposefully left it there. Or do you think it would be better in the Vortex child class? I am not a fan of that since `q` is a parameter of this class here, but of course it depends on the actual coronagraph it's used with.

Reply:
I think this is okay. It helps people understand what the correct order is for the parameter.